miden-lib/asm/kernels/transaction/lib/account.masm

```diff
@@ -349,6 +365,24 @@ export.get_storage_slot_type_info
     exec.memory::get_acct_storage_slot_type_data u32split
 end
 
+#! Offsets provided storage slot index for storage access
+#!
+#! Panics:
+#!   - slot index is out of bounds
+#!   - computed index is out of bounds
+#!
+#! Stack: [storage_offset, slot_index]
+#! Output: [offset_slot_index]
+export.offset_storage_index
+    # verify that slot_index is in bounds
+    dup.1 exec.get_max_storage_slot_index lt assert.err=ERR_STORAGE_SLOT_INDEX_OUT_OF_BOUNDS
+    # => [storage_offset, slot_index]
+
+    # verify that the computed offset_slot_index is in bounds
+    add dup exec.get_max_storage_slot_index lt assert.err=ERR_STORAGE_SLOT_INDEX_OUT_OF_BOUNDS
```

Comment:
nit: since `MAX_STORAGE_SLOT_INDEX` is defined locally, I'd probably use just `push.MAX_STORAGE_SLOT_INDEX` instead of calling `get_max_storage_slot_index` procedure.

This would also mean that we probably don't even need to define `get_max_storage_slot_index` (I don't think it is used outside of this file).